editor/execution.py

```diff
@@ -61,3 +68,7 @@ def string_exec(strings, out, global_frame=None):
             if not log.logger.fragile:
                 log.logger.out("Time limit exceeded.")
         log.logger.new_expr()
+
```

Comment:
Why is this an error?

Reply:
Otherwise, the console will just go blank with no indication of what happened. This is just a way of pushing a message. `ParseError`s also indicate that a persistent environment has not been set up, so the user should not be able to continue to interact.

Reply:
Could we just execute return `undefined` if the file is empty?

Reply:
That would make `undefined` the first statement in the debugger. Is that OK?

Reply:
Oh I see, it's fine I guess yeah. Makes more sense than crashing

Reply:
Wait, so do you want me to make `undefined` the first statement, or should I keep it how it is now?

Reply:
would prefer `undefined`

Reply:
That's how it's done now after the latest commit.